### PR TITLE
[Remove] Types from PutIndexTemplateRequest and builder to reduce mapping to a string

### PR DESCRIPTION
**Pull request by dreamer-89**
Signed-off-by: Suraj Singh <surajrider@gmail.com>

### Description
With type removal from codebase, this change prevents `PutIndexTemplateRequest` and corresponding builder to hold multiple mapping types. It change internal representation of mapping from maps to String and a subsequent change of https://github.com/opensearch-project/OpenSearch/pull/2497

Related: #1940

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/internalClusterTest/java/org/opensearch/cluster/SimpleClusterStateIT.java [hunk 1]
```diff
@@ -166,11 +166,10 @@ public void testIndexTemplates() throws Exception {
             .preparePutTemplate("foo_template")
             .setPatterns(Collections.singletonList("te*"))
             .setOrder(0)
-            .addMapping(
-                "type1",
+            .setMapping(
                 XContentFactory.jsonBuilder()
                     .startObject()
-                    .startObject("type1")
+                    .startObject("_doc")
```
Comment (reta) [inline server/src/internalClusterTest/java/org/opensearch/cluster/SimpleClusterStateIT.java:172]: I think the `_doc` part is not needed

Comment (dreamer-89) [reply server/src/internalClusterTest/java/org/opensearch/cluster/SimpleClusterStateIT.java:172]: Thank you @reta for your comment. Yes, it looks like _doc is not needed.

I will remove all occurrences of _doc added as part of this commit.

**Comment by dreamer-89** [comment]
Existing gradle check is failing due to parsing issue. Probable bug in logic change in `PutIndexTemplateRequest.toXContent` method
```
  2> REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.action.admin.indices.template.put.PutIndexTemplateRequestTests.testFromXContent" -Dtests.seed=B211A97EE49E7566 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=sl-SI -Dtests.timezone=Asia/Aqtau -Druntime.java=17
  2> java.lang.NullPointerException: Cannot invoke "String.length()" because "content" is null
        at __randomizedtesting.SeedInfo.seed([B211A97EE49E7566:93799C666D3995BC]:0)
        at com.fasterxml.jackson.core.JsonFactory.createParser(JsonFactory.java:1158)
        at org.opensearch.common.xcontent.json.JsonXContent.createParser(JsonXContent.java:97)
        at org.opensearch.action.admin.indices.template.put.PutIndexTemplateRequest.toXContent(PutIndexTemplateRequest.java:523)
```

File: server/src/internalClusterTest/java/org/opensearch/cluster/SimpleClusterStateIT.java (hunk 1 above)
Comment (reta) [reply server/src/internalClusterTest/java/org/opensearch/cluster/SimpleClusterStateIT.java:172]: :+1:

**Review by reta — APPROVED** [review]

**Comment by dreamer-89** [comment]
start gradle check

**Comment by dreamer-89** [comment]
```
**REPRODUCE WITH: ./gradlew ':qa:remote-clusters:integTest' --tests "org.opensearch.cluster.remote.test.RemoteClustersIT.testHAProxyModeConnectionWorks" -Dtests.seed=822F6BCFE6BD6C51 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=ar-KW -Dtests.timezone=Asia/Colombo -Druntime.java=17

org.opensearch.cluster.remote.test.RemoteClustersIT > testHAProxyModeConnectionWorks FAILED
    java.lang.AssertionError
        at __randomizedtesting.SeedInfo.seed([822F6BCFE6BD6C51:8559BE04DD3B1A0C]:0)
        at org.junit.Assert.fail(Assert.java:87)
        at org.junit.Assert.assertTrue(Assert.java:42)
        at org.junit.Assert.assertTrue(Assert.java:53)
        at org.opensearch.cluster.remote.test.RemoteClustersIT.testHAProxyModeConnectionWorks(RemoteClustersIT.java:125)**
```

**Comment by dreamer-89** [comment]
start gradle check

**Comment by dreamer-89** [comment]
@nknize : I would also request your review on this PR as it involves changes around serialisation. Can you please have a look ?

**Review by nknize — APPROVED** [review]
LGTM!

**Pull request merged**